pkg/chunked/storage_linux.go

```diff
@@ -1024,6 +1074,21 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions) (gra
 			continue
 		}
 
+		found, dstFile, _, err = findFileInOSTreeRepos(&r, ostreeRepos, dirfd, useHardLinks)
```

Comment:
Can found be false and dstFIle != nil?  Or found be true and dstFile be nil?
If not then the function should only return dstFile.

Reply:
yes, it can return `found == true`, and `dstNil == nil` when doing hard links deduplication so the destination file is not overwritten.

Reply:
How about found ==false dstNil != nil?

Reply:
no that is not possible.  If the file is not found then the destination is not opened

Reply:
It would make this clearer if it said
```
if found {
	if dstFile != nil {
			if err := setFileAttrs(dstFile, mode, &r, options); err != nil {
				dstFile.Close()
				return output, err
			}
			dstFile.Close()
		}
		
		continue
}
```

Reply:
I've simplified it further and pushed a new version